### PR TITLE
Add ACM certificate for `rules.vm.cyber.dhs.gov`

### PR DESCRIPTION
**Pull request by dav3r**
#  #

## 🗣 Description ##

This PR creates and validates an ACM certificate for `rules.vm.cyber.dhs.gov`.  It also adds a new policy and role that is needed in order to manage ACM resources.

## 💭 Motivation and context ##

We want to migrate our egress IP rules information from https://rules.ncats.cyber.dhs.gov to a new URL https://rules.vm.cyber.dhs.gov (which is not yet set up).  Creating the certificate for `rules.vm.cyber.dhs.gov` is a necessary precursor for that migration.

This is related to the changes for:
* https://github.com/cisagov/cool-system/issues/208
* https://github.com/cisagov/cyhy_amis/issues/284

## 🧪 Testing ##

I applied this Terraform and confirmed that the certificate for `rules.vm.cyber.dhs.gov` was created as expected.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

**Review by jsf9k — APPROVED** [review]
LGTM!

File: acm_rules_vm.tf [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    for dvo in aws_acm_certificate.rules.domain_validation_options : dvo.domain_name => {
+      name   = dvo.resource_record_name
+      record = dvo.resource_record_value
+      type   = dvo.resource_record_type
+    }
+  }
+
+  allow_overwrite = true
+
+  depends_on = [aws_acm_certificate.rules]
```
Comment (mcdonnnj) [inline acm_rules_vm.tf]: Is this strictly necessary? Since the resource is used in the `for_each` it should be an implicit dependency right?

**Review by mcdonnnj — APPROVED** [review]
This looks pretty solid. It might be overkill but as a thought it would be nice if some of the comments/descriptions expanded ACM out to AWS Certificate Manager just to improve documentation a bit.

File: acm_rules_vm.tf (hunk 1 above)
Comment (dav3r) [reply acm_rules_vm.tf]: Good catch - I cribbed that code from somewhere else and didn't think to remove that unnecessary `depends_on`.  That has been rectified in https://github.com/cisagov/cool-dns-cyber.dhs.gov/pull/69/commits/ec0bf559a2a986be3a523a6bd79060c8a7377795.

**Comment by dav3r** [comment]
> It might be overkill but as a thought it would be nice if some of the comments/descriptions expanded ACM out to AWS Certificate Manager just to improve documentation a bit.

Thanks for the suggestion - please take a look at https://github.com/cisagov/cool-dns-cyber.dhs.gov/pull/69/commits/9f889e9bbdf466c895a1503a3312d26b93349c76.

**Pull request merged**